gateway-provider-security-authz-acls/src/main/java/org/apache/knox/gateway/filter/AclsAuthorizationFilter.java

```diff
@@ -128,19 +124,19 @@ protected boolean enforceAclAuthorizationPolicy(ServletRequest request,
     boolean groupAccess = false;
     boolean ipAddrAccess;
 
-    Subject subject = Subject.getSubject(AccessController.getContext());
-    Principal primaryPrincipal = (Principal)subject.getPrincipals(PrimaryPrincipal.class).toArray()[0];
-    log.primaryPrincipal(primaryPrincipal.getName());
-    Object[] impersonations = subject.getPrincipals(ImpersonatedPrincipal.class).toArray();
-    if (impersonations.length > 0) {
-      log.impersonatedPrincipal(((Principal)impersonations[0]).getName());
-      userAccess = checkUserAcls((Principal)impersonations[0]);
+    final Subject subject = SubjectUtils.getCurrentSubject();
```

Comment:
Could this be simplified to
`
effectivePrincipalName = SubjectUtils.getCurrentEffectivePrincipalName();
log.effectivePrincipal(effectivePrincipalName);
userAccess = checkUserAcls(effectivePrincipalName);
`

Reply:
Nice catch; I'll fix it soon.

Reply:
Done.